### PR TITLE
merge bitcoin#13671, #17405, #18786, #18792, #20067: Deboostification

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

File: src/wallet/walletdb.cpp [hunk 1]
```diff
@@ -667,9 +665,6 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
         pwallet->nKeysLeftSinceAutoBackup = pwallet->KeypoolCountExternalKeys();
         pwallet->WalletLogPrintf("nKeysLeftSinceAutoBackup: %d\n", pwallet->nKeysLeftSinceAutoBackup);
     }
```
Comment (UdjinM6) [inline src/wallet/walletdb.cpp:667]: 18792:
```suggestion
```

File: src/wallet/walletdb.cpp [hunk 2]
```diff
@@ -667,9 +665,6 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
...
-    catch (const boost::thread_interrupted&) {
-        throw;
-    }
     catch (...) {
```
Comment (UdjinM6) [inline src/wallet/walletdb.cpp:668]: 18792:
```suggestion
    } catch (...) {
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
13734 was already merged recently
18234 missing earlier scheduler related backports. This part is critical, would prefer to postpone it for now and backport them in the right order.

File: src/test/util_tests.cpp [hunk 3]
```diff
@@ -1289,4 +1289,43 @@ BOOST_AUTO_TEST_CASE(test_tracked_vector)
...
+    BOOST_CHECK_EQUAL(ToLower(0), 0);
+    BOOST_CHECK_EQUAL(ToLower(255), 255);
+
+    std::string testVector;
+    Downcase(testVector);
+    BOOST_CHECK_EQUAL(testVector, "");
+
+    testVector = "#HASHTAG";
+    Downcase(testVector);
+    BOOST_CHECK_EQUAL(testVector, "#hashtag");
```
Comment (PastaPastaPasta) [inline src/test/util_tests.cpp:1307]: Why'd you change this? Should be #hodl

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
18792 should drop `#include <boost/thread.hpp>` in `src/wallet/db.cpp` like in the original PR. 18792 requires 17405 (which in its turn requires 11911) to be compilable (I hope), pls see https://github.com/UdjinM6/dash/commits/pr4246

EDIT: Hmm.. tests failed, so I guess there smth else is missing (or maybe I backported them incorrectly)

**Comment by PastaPastaPasta** [comment]
TBH I would just prefer to maybe mark 18792 as partial and merge as is (unless either of you want to get that branch working)

**Comment by UdjinM6** [comment]
> (unless either of you want to get that branch working)

working on it :)

**Comment by UdjinM6** [comment]
Decided to backport 11911 in a separate PR #4263 since it's kind of unrelated to "deboostification". Once it's merged backporting 17405 and 18792 should be trivial.

**Comment by kittywhiskers** [comment]
18792 originally dropped `boost/thread.hpp` but apparently it was still required so I modified the commit and placed it back in

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
> 18792 originally dropped `boost/thread.hpp` but apparently it was still required so I modified the commit and placed it back in

Pls revert this to make 18792 complete. Rebasing and adding 17405 (e.g. 1722e83588d41d9f3455986fc52d058d68300a33) should fix the issue.

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**